lib/addons/velero/index.ts

```diff
@@ -0,0 +1,259 @@
+                    }
+                }             
+            }
+        };
+
+        const values = merge(props.values, valueVariable.values) ?? {}; 
+ 
+        this.chartNode = cluster.addHelmChart("velero-addon", {
+            chart: "velero",
+            repository: "https://vmware-tanzu.github.io/helm-charts/",
+            release: Constants.SSP_ADDON,
+            namespace: veleroNamespace,
+            version: props.version,
+            values: values
+        });
+        return this.chartNode;
+    }
+
+    /**
+     * Return S3 Bucket
+     * @param clusterInfo 
+     * @param id S3-Bucket-Postfix 
+     * @param existingBucketName exiting provided S3 BucketName if it exists 
+     * @returns the existing provided S3 bucket name or the newly created S3 bucket name
+     */
+    protected returnS3BucketName(clusterInfo: ClusterInfo, id: string, existingBucketName: null|string ): string {
+        if (!existingBucketName){
+            const bucket = new s3.Bucket(clusterInfo.cluster, "velero-${id}", {
+                encryption: s3.BucketEncryption.KMS_MANAGED, // Velero Known bug for support with S3 with SSE-KMS with CMK, thus it does not support S3 Bucket Key: https://github.com/vmware-tanzu/helm-charts/issues/83
+                blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL, // Block Public Access for S3
+                publicReadAccess: false,
+                enforceSSL: true // Encryption in Transit
+            });
+            return bucket.bucketName;
+        }
+        else {
+            return existingBucketName;
+        }
+    }
+
+    /**
+     * Return Velero Namespace where Velero will be installed onto
+     * @param clusterInfo
+     * @param defaultName the Default Namespace for Velero if nothing specified 
+     * @param namespace
+     * @returns the namespace created or existed.
+     */
+    protected createNamespaceIfNeeded(clusterInfo: ClusterInfo, defaultName: string, namespace?: {name: string, create: boolean}): string {
+        // Create Namespace if namespace is not explicied defined.
+        if (namespace){
+            // Create Namespace if the "create" option is true
+            if (namespace.create) {
+                createNamespace(namespace.name, clusterInfo.cluster);
+            }
+            // If the "create" option if false, then namespace will not be created, return namespace.name
+            return namespace.name;
+        }
+        else{
+            return defaultName; // initial value of veleroNamespace
+        }
+    }
+
+    /**
+     * Return Velero Namespace where Velero will be installed onto
+     * @param clusterInfo
+     * @param id
+     * @param namespace Velero namespace name
+     * @param s3BucketName the S3 BucketName where Velero will stores the backup onto
+     * @returns the service Account Name
+     */
+    protected createIRSA(clusterInfo: ClusterInfo, id: string, namespace: string, s3BucketName: string): string {
```

Comment:
You will need to return the actual ServiceAccount as opposed to string to setup dependencies.

Reply:
IRSA stands for IAM Roles for Service Accounts. So the method name is `createIAMRolesForServiceAccounts` which does not sound right, although I understand what you mean. Consider renaming to `createServiceAccountWithIamRoles` or just `createServiceAccount`.

Reply:
resolved.